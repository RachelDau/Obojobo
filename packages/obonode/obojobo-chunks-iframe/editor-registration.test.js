import IFrame from './editor-registration'

const IFRAME_NODE = 'ObojoboDraft.Chunks.IFrame'

describe('IFrame editor', () => {
	test('plugins.isVoid to call next if not IFrame', () => {
		const next = jest.fn()
		IFrame.plugins.isVoid({}, {}, next)
		expect(next).toHaveBeenCalled()
	})

	test('plugins.isVoid to return true if the node is an IFrame', () => {
		const node = { type: IFRAME_NODE }
		expect(IFrame.plugins.isVoid(node, {}, jest.fn())).toEqual(true)
	})

	test('plugins.renderNode renders an IFrame when passed', () => {
		const props = {
			attributes: { dummy: 'dummyData' },
			node: {
				type: IFRAME_NODE,
				data: {
					get: () => {
						return {}
					}
				}
			}
		}

		expect(IFrame.plugins.renderNode(props, null, jest.fn())).toMatchSnapshot()
	})
})
