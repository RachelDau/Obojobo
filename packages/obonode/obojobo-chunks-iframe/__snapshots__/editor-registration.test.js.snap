// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IFrame editor plugins.renderNode renders an IFrame when passed 1`] = `
<SlateWrapper
  attributes={
    Object {
      "dummy": "dummyData",
    }
  }
  dummy="dummyData"
  node={
    Object {
      "data": Object {
        "get": [Function],
      },
      "type": "ObojoboDraft.Chunks.IFrame",
    }
  }
/>
`;
